sql/src/main/java/io/crate/analyze/Analyzer.java

```diff
@@ -485,18 +492,28 @@ public AnalyzedStatement visitCreateIngestRule(CreateIngestRule node, Analysis c
 
         @Override
         public AnalyzedStatement visitCreateView(CreateView createView, Analysis analysis) {
-            return createViewAnalyzer.analyze(
-                createView, analysis.transactionContext(), analysis.sessionContext().defaultSchema());
+            return createViewAnalyzer.analyze(createView, analysis.transactionContext());
         }
 
         @Override
         public AnalyzedStatement visitDropView(DropView dropView, Analysis analysis) {
             // No exists check to avoid stale clusterState race conditions
-            String defaultSchema = analysis.sessionContext().defaultSchema();
             ArrayList<RelationName> views = new ArrayList<>(dropView.names().size());
+            ArrayList<RelationName> missing = new ArrayList<>();
             for (QualifiedName qualifiedName : dropView.names()) {
-                views.add(RelationName.of(qualifiedName, defaultSchema));
+                try {
+                    views.add(schemas.resolveView(qualifiedName, analysis.sessionContext().searchPath()).v2());
+                } catch (RelationUnknown e) {
+                    if (!dropView.ifExists()) {
+                        missing.add(RelationName.of(qualifiedName, analysis.sessionContext().searchPath().currentSchema()));
+                    }
+                }
             }
+
+            if (!missing.isEmpty()) {
+                throw new RelationsUnknown(missing);
```

Comment:
if we throw an exception if one of the relations is unknown, how is it published to the user that the known ones are dropped successfully?

Reply:
It's a matter of the user deducting that the _others_ were successful. This was the existing behaviour with this validation being performed at execution times (it's still there to catch the cases where views went away between the analyze and execute phase). This is not a behaviour change.